worker/zero_proxy.go

```diff
@@ -5,10 +5,14 @@ import (
 
 	"github.com/dgraph-io/dgraph/conn"
 	"github.com/dgraph-io/dgraph/protos/pb"
+	"github.com/pkg/errors"
 	"google.golang.org/grpc"
 )
 
 func forwardAssignUidsToZero(ctx context.Context, in *pb.Num) (*pb.AssignedIds, error) {
+	if in.Type != pb.Num_UID {
```

Comment:
Add JWT check here. Just check if the user belongs to some valid namespace. Pass the namespace in context to zero.

Rate limiting in Zero. Probably add a delay by ourselves on zero.
For namespace 0, no limits. Default is infinite.